megamek/src/megamek/client/ui/swing/boardview/BoardView.java

```diff
@@ -5798,6 +5695,146 @@ public String getHexTooltip(MouseEvent e) {
         return txt.toString();
     }
 
+    public void appendTerrainTooltip(StringBuffer txt, Hex mhex, String bgColor) {
```

Comment:
If this are public, we don't know where the call will come from - so we probably want to put a null check and early cutout on mhex here.

Reply:
Looking at it, getHexTooltip shouldn't be public,  so I re-combined it into the mouse event func above. The public parts (appendTerrainTooltip, etc) I added javadoc and null checks.
Elevation is shown if non-zero. This was the default behavior in the tooltip

Reply:
I'll look into using the skin background. It might be a big enough project to move into another RFE though.

Reply:
Skin texture bg added